src/nupic/algorithms/SpatialPooler.hpp

```diff
@@ -42,6 +40,11 @@ namespace nupic {
 namespace algorithms {
 namespace spatial_pooler {
 
+// force small data types
+using Real = Real32;
+using UInt = UInt32;
+using Int = Int32; 
```

Comment:
How is this useful on per file basis? I think the idea of `Real` was exactly to avoid type conversions, so the whole codebase would use Real (etc) and then in one file you set `Real = Real32` for all the code. Isn't that what you want? 
Using Real64/32 is only for explicit cases.

Reply:
Actually, in Types.hpp all Real's are set to Real32. (Unless you set double precision)
But it is not obvious looking at the code how big it is.  I personally would like to see Real32 everywhere....or even "float" for floating point numbers.

There are reasons for being 32 bit.  One example are the Links.  The links between plugins uses a fixed with of 32bits per element.  This is 'typeless' because in reality it could be any 32bit type so the buffer is defined as an array of char.  Buffer size then is elements * element size.   It then does a char array copy. The bigger the size of the element the longer it takes to copy.  There are other places that also use byte arrays to hold data of any type.

Since the values are always 0, non-0  in SDR's we could be consistent and use Byte arrays everywhere (for SDR's) and save conversion and copy time.  For links that are not SDR's we could still handle because the link is passing ArrayBuffer objects which know the type and length.  But for SDR array's we could shorten the copy time by using Byte.

Anyway, that is my soapbox.  Perhaps we should defer this for a separate PR and do some research into how best to handle SDR's as has been suggested in #109.